volatility3/framework/plugins/linux/capabilities.py

```diff
@@ -0,0 +1,228 @@
+        )
+        vol2_last_cap = extensions.kernel_cap_struct.get_last_cap_value()
+        if kernel_cap_last_cap > vol2_last_cap:
+            vollog.warning(
+                "Developers: The supported Linux capabilities of this plugin are outdated for this kernel"
+            )
+
+    @staticmethod
+    def _decode_cap(cap: interfaces.objects.ObjectInterface) -> str:
+        """Returns a textual representation of the capability set.
+        The format is a comma-separated list of capabilitites. In order to
+        summarize the output and if all the capabilities are enabled, instead of
+        the individual capabilities, the special name "all" will be shown.
+
+        Args:
+            cap: Kernel capability object. Usually a 'kernel_cap_struct' struct
+
+        Returns:
+            str: A string with a comma separated list of decoded capabilities
+        """
+        if isinstance(cap, renderers.NotAvailableValue):
+            return cap
+
+        cap_value = cap.get_capabilities()
+        if cap_value == 0:
+            return "-"
+
+        CAP_FULL = 0xFFFFFFFF
+        if cap_value == CAP_FULL:
+            return "all"
+
+        return ", ".join(cap.enumerate_capabilities())
+
+    @classmethod
+    def get_task_capabilities(cls, task: interfaces.objects.ObjectInterface) -> Dict:
+        """Returns a dict with the task basic information along with its capabilities
+
+        Args:
+            task: A task object from where to get the fields.
+
+        Returns:
+            dict: A dict with the task basic information along with its capabilities
+        """
+        task_cred = task.real_cred
+        fields = {
+            "common": [
+                utility.array_to_string(task.comm),
+                int(task.pid),
+                int(task.tgid),
+                int(task.parent.pid),
+                int(task.cred.euid),
+            ],
+            "capabilities": [
+                task_cred.cap_inheritable,
+                task_cred.cap_permitted,
+                task_cred.cap_effective,
+                task_cred.cap_bset,
+            ],
+        }
+
+        # Ambient capabilities were added in kernels 4.3.6
+        if task_cred.has_member("cap_ambient"):
+            fields["capabilities"].append(task_cred.cap_ambient)
+        else:
+            fields["capabilities"].append(renderers.NotAvailableValue())
+
+        return fields
+
+    def get_tasks_capabilities(
+        self, tasks: List[interfaces.objects.ObjectInterface]
+    ) -> Iterable[Dict]:
+        """Yields a dict for each task containing the task's basic information along with its capabilities
+
+        Args:
+            tasks: An iterable with the tasks to process.
+
+        Yields:
+            Iterable[Dict]: A dict for each task containing the task's basic information along with its capabilities
+        """
+        for task in tasks:
+            if task.is_kernel_thread:
```

Comment:
Could we get a quick comment just to explain why we're filtering out kernel thread here please?